### PR TITLE
Add cast_waggle function to DRCA

### PR DESCRIPTION
**Pull request by Tarhish**
The method takes settings and a waggle name, then casts it in an identical manner to ;buff.

### CODE REVIEW

File: common-arcana.lic [hunk 1]
```diff
@@ -640,4 +640,23 @@ module DRCA
 
     crafting_prepare_spell(training_spells[needs_training], settings)
   end
+
+  def cast_waggle(settings, waggle_sets_name = 'default')
```
Comment (rpherbig) [inline common-arcana.lic]: 💡 I don't like 'waggle' and I think it's overall been more confusing than helpful.

File: common-arcana.lic [hunk 2]
```diff
@@ -640,4 +640,23 @@ module DRCA
...
+    return unless settings.waggle_sets[waggle_sets_name]
+
+    spells = settings.waggle_sets[waggle_sets_name]
+
+    if DRStats.barbarian?
```
Comment (rpherbig) [inline common-arcana.lic]: ❓ Did you confirm that `DRStats` works here in a common-* helper?

File: common-arcana.lic (hunk 1 above)
Comment (rpherbig) [inline common-arcana.lic]: 💡 I'm not a fan of the default value. I think `buff` should be responsible for figuring out how to interpret the user's args. Thoughts?

File: common-arcana.lic [hunk 3]
```diff
@@ -640,4 +640,23 @@ module DRCA
...
+      start_barb_abilities(spells, settings)
+    elsif DRStats.thief?
+      start_khris(spells, settings)
+    else
+      spells.each_value do |value|
```
Comment (rpherbig) [inline common-arcana.lic]: 💡 Could we do something like this?

```ruby
spells.values
.select { |spell| spell['use_auto_mana'] }
.each { |spell| check_discern(spell, settings) }
```

File: common-arcana.lic (hunk 1 above)
Comment (Tarhish) [reply common-arcana.lic]: I think you're right,  and new lichers have no idea what waggles are.  They sound like something irrelevant just by their name.  But if that is their name I'd like to stick to it, unless this is the time we add a transform to dependency that maps wagglesets to a different name and begin the long process of transitioning away from it?  I'd do it, but that's a bigger codebase change. 

Alternately I could make this more generic and accept any hash of spells.

File: common-arcana.lic (hunk 3 above)
Comment (Tarhish) [reply common-arcana.lic]: I think it should work -- I'll try it out.

File: common-arcana.lic (hunk 1 above)
Comment (rpherbig) [reply common-arcana.lic]: I think the only place we have to use 'waggle' is in the setting name itself, since we haven't fixed that yet.

Comment (Tarhish) [reply common-arcana.lic]: Done and done

Comment (Tarhish) [reply common-arcana.lic]: I got distracted by other things, but I'm back to this.  Still, I'll look at the wider scale some other time maybe.

**Comment by Tarhish** [comment]
I ran through those changes to make sure they work the same way, and it's been successful.

**Pull request merged**